File: Leetcode/24. Swap Nodes in Pairs.py
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None:
            return head
        if head.next is None:
            return head
        a = head.val
        b = head.next.val
        head.next.val = a
        head.val = b
        self.swapPairs(head.next.next)
        return head

File: Leetcode/1-999/24. Swap Nodes in Pairs.java
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head == null) return head;
        if(head.next == null) return head;

        ListNode a = head;
        ListNode b = head.next;
        a.next = b.next;
        b.next = a;
        head = b;

        head.next.next = swapPairs(head.next.next);
        return head;
    }
}
